src/emulator/dataconnect/pgliteServer.ts

```diff
@@ -39,13 +39,22 @@ export class PostgresServer {
           // pglite wrongly sends.
           return extendedQueryPatch.filterResponse(data, result);
         },
+
+        async onAuthenticated() {
+          // Every time we see a new authentication exchange, we need to throw away previously prepared statements -
+          // Clients may not know of these statements, and may attempt to reuse the same name.
+          await db.query("DEALLOCATE ALL");
```

Comment:
I still don't think this is quite right -- what if there are two clients connected at the same time? Like, one tries to prepare a statement and then the other one authenticates. 

Shouldn't we be deallocating when clients disconnect instead? Prepared statements should be per connection